routing/routing.xml

```diff
@@ -23,29 +23,226 @@
-	        value of 1 means way passable.
-	     2) <way attribute="speed" type="speed">
-	        Defines speed limits on roads in km/h (if it is not specified on the road via maxspeed tag)
-	     3) <way attribute="priority">
-	     	Priority is a preference multiplicator for the road type (consider it is an accelerator of the road)
-	     4) <point attribute="obstacle">
-	        Obstacle defines a point of the road and whether it can be passed by the vehicle.
-	        Penalty is measured in seconds, a value of -1 means the obstacle cannot be passed and a value of 0 means "no penalty"!
-	        For routing engine.
-             5) <point attribute="obstacle_time">
-                Penalty for crossing a feature on the road. It is measured in seconds.
-                Only for visual effect (ETA).
-	     		
-	   The order of specialization matters only how it was supplied :
-	      short_way, avoid_ferries, avoid_toll, avoid_motorway, avoid_unpaved
-	   
-	   Conditional values (based on parameters) can be specified as:
-	      <select value="x" t="y" v="z">
-                <if param="parameter id"/>
-              </select>
+	
+		If you want to tweak the routing engine, read this carefully.
+		
+    Deploying your own routing.xml
+    ==============================
+		
+		When you finish editing this document (or want to test on your Android device), copy this file over to:
+		sdcard/Android/data/net.osmand.plus/files/
+		
+		Restart OsmAnd (e.g. by swiping it away in the recent app view) to activate the new routing.xml;
+		
+    Overview of the document
+    ------------------------
+	
+		The document contains many big blocks, each describing a single profile.
+		OsmAnd has three default profiles: car, bicycle and foot.
+		
+		Some others are described here as well but are hidden by default.
+		These profiles can be enabled using the 'OsmAnd development plugin'
+		
+	## Routing profile 
+		
+		For starters, a profile can have a few parameter declarations, to finetune the profile:
+		
+		<parameter name="some_toggle" ... />
+		<parameter name="some_other_preference" ... />
+		
+        (Details on parameters are given later on)
+		
+		A profile block contains at least the following segments:
+		1) access                 (which determines if a road can be taken)
+   		2) speed                  (which determines what max speed a road allows)
+    	
+    	A profile block might also contain following segments:
+    	3) oneway                 (which determines in which direction a road can be taken)
+    	4) priority               (which determines how nice the road is to drive on)
+    	5) obstacle               (which determines the delay of an obstacle, in seconds for the routing engine)	
+    	6) obstacle_time          (which determines the delay of an obstacle, in seconds for the human)
+    	7) penalty_transition     (which determines the penalty of transitioning from one road surface to another)
```

Comment:
That's not true. penalty_transition is penalty for the exit from a high-level road to a low-level road (primary to primary_link e.g.)